### PR TITLE
Merge dockerhub automated build version

### PR DESCRIPTION
**Pull request by sychan**
This version should build a smaller standalone shock image automatically with docker hub. Also has docker-compose setup and initial make target that will bring up shock + mongo for testing within Travis. Actual tests are still to be written.

### CODE REVIEW

File: shock-server.cfg [hunk 1]
```diff
@@ -0,0 +1,56 @@
+[Address]
+# IP and port for api
+# Note: use of port 80 may require root access
+# 0.0.0.0 will bind Shock to all IP's
+api-ip=0.0.0.0
+api-port=7044
+
+[Admin]
+email=kkeller@lbl.gov
```
Comment (kkellerlbl) [inline shock-server.cfg]: Can we remove my email address from the example cfg file?

File: shock-server.cfg [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+[Mongodb]
+# Mongodb configuration
+# Hostnames and ports hosts=host1[,host2:port,...,hostN]
+hosts=mongo
+database=shock
+user=
+password=
+gomaxprocs=4
+admin-email=sysadmins@kbase.us
```
Comment (kkellerlbl) [inline shock-server.cfg]: These admin directives look like they're in the wrong place (even though they are actually in the existing CI shock.cfg file).

File: shock-server.cfg [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+[Anonymous]
+# Controls an anonymous user's ability to read/write
+# values: true/false
+read=true
+write=true
+create-user=true
+
+[Auth]
+; for auth2
+globus_token_url=https://ci.kbase.us/services/auth/api/legacy/globus/goauth/token?grant_type=client_credentials
```
Comment (kkellerlbl) [inline shock-server.cfg]: Would Travis use these services, or would it spin up its own auth2 service for testing?

**Review by kkellerlbl — CHANGES_REQUESTED** [review]
The important change is removing my email.  Otherwise I think it looks ok.

**Comment by sychan** [comment]
@kkellerlbl All request changes made. Let me know if you think this is good to go so that I can test the automated build

**Review by kkellerlbl — APPROVED** [review]
Looks okay.

**Pull request merged**